sms_sponsorship/models/recurring_contract.py

```diff
@@ -179,6 +179,13 @@ def contract_waiting(self):
         self._post_payment_first_month()
         return True
 
+    @api.multi
+    def contract_active(self):
+        super().contract_active()
+        if self.sms_request_id:
+            self.sms_request_id.state = "step2"
```

Comment:
This should be done in `contract_waiting` method instead of `contract_active`.

```suggestion
        self.mapped("sms_request_id").write({"state": "step2"})
```

Reply:
Changes are done